src/sql/magic_cmd.py

```diff
@@ -65,8 +66,118 @@ def execute(self, line="", cell="", local_ns=None):
+                required=False,
+            )
+            parser.add_argument(
+                "-loe",
+                "--less-than-or-equal",
+                type=str,
+                help="Less than or equal to a certain number.",
+                required=False,
+            )
+            parser.add_argument(
+                "-nn",
+                "--no-nulls",
+                help="Returns rows in specified column that are not null.",
+                action="store_true",
+            )
+
+            args = parser.parse_args(others)
+            if args.greater and args.greater_or_equal:
+                return ValueError(
+                    "You cannot use both greater and greater "
+                    "than or equal to arguments at the same time."
+                )
+            elif args.less_than and args.less_than_or_equal:
+                return ValueError(
+                    "You cannot use both less and less than "
+                    "or equal to arguments at the same time."
+                )
+
+            conn = sql.connection.Connection.current.session
+            result_dict = run_each_individually(args, conn)
+
+            if len(result_dict.keys()):
+                print(
+                    "Test failed. Returned are samples of the failures from your data:"
+                )
+                return result_dict
+            else:
+                print("All requirements were met. Test was succesful.")
+                return True
+
+        raise UsageError(
+            f"%sqlcmd has no command: {cmd_name!r}. "
+            "Valid commands are: 'tables', 'columns'"
+        )
+
+
+def run_each_individually(args, conn):
+    base_query = select("*").from_(args.table)
+    storage = {}
+
+    if args.greater:
+        where = condition(args.column + ">" + args.greater)
+        current_query = base_query.where(where).sql()
+
+        res = conn.execute(current_query).fetchone()
+
+        if res is not None:
+            storage["greater"] = res
+    if args.greater_or_equal:
+        where = condition(args.column + ">=" + args.greater_or_equal)
+
+        current_query = base_query.where(where).sql()
+
+        res = conn.execute(current_query).fetchone()
+        if res is not None:
+            storage["greater_or_equal"] = res
+    if args.less_than_or_equal:
+        where = condition(args.column + "<=" + args.less_than_or_equal)
+        current_query = base_query.where(where).sql()
+
+        res = conn.execute(current_query).fetchone()
+        if res is not None:
+            storage["less_than_or_equal"] = res
+    if args.less_than:
+        where = condition(args.column + "<" + args.less_than)
+        current_query = base_query.where(where).sql()
+
+        res = conn.execute(current_query).fetchone()
+        if res is not None:
+            storage["less_than"] = res
+    if args.no_nulls:
+        current_query = base_query.sql() + " WHERE " + args.column + " IS NULL"
```

Comment:
Should use [WHERE Class](https://sqlglot.com/sqlglot/expressions.html#Where) instead of concatenating the "WHERE" string here

Reply:
for some reason I couldn't get it to work with the WHERE class. not sure if it supports `.where($variable + "IS NULL)`

Reply:
Yes, it supports something like this
```
from sqlglot import select, condition
where = condition("x is NULL")
select("*").from_("y").where(where).sql()
```

So in your case
```
where = condition("{} is NULL".format(args.column))
current_query = base_query.where(where).sql()
```

Reply:
sounds good, that's pretty elegant.